fun_certificates/views.py

```diff
@@ -12,10 +12,21 @@
 
 def short_cert_url(request, encoded_cert_id):
     """Retrieve GeneratedCertificate from its encoded short id and returns
-    edX's view content for it."""
-    cert = get_object_or_404(GeneratedCertificate,
-        id=cert_id_decode(settings.SECRET_KEY, encoded_cert_id),
-        status=CertificateStatuses.downloadable,
-        mode=GeneratedCertificate.MODES.verified)
+    edX's view content for it.
+
+    We have changed the SECRET_KEY, we need to check its previous value to access previously generated short urls.
+    """
+
+    # check the current secret key
+    try:
+        cert = GeneratedCertificate.objects.get(id=cert_id_decode(settings.SECRET_KEY, encoded_cert_id),
+            status=CertificateStatuses.downloadable,
+            mode=GeneratedCertificate.MODES.verified)
+    except GeneratedCertificate.DoesNotExist:
+        # check empty secret_key
+        cert = get_object_or_404(GeneratedCertificate,
+               id=cert_id_decode("", encoded_cert_id),
+            status=CertificateStatuses.downloadable,
+            mode=GeneratedCertificate.MODES.verified)
```

Comment:
Ou encore : 

    cert_id = cert_id_decode(settings.SECRET_KEY, encoded_cert_id) or cert_id_decode("", encoded_cert_id)
    if cert_id is None:
        raise Http404
    cert = get_object_or_404(
        GeneratedCertificate,
         id=cert_id,
         status=CertificateStatuses.downloadable,
         mode=GeneratedCertificate.MODES.verified
    )